can_tools/scrapers/puppet.py

```diff
@@ -144,9 +166,16 @@ async def _post_download(self, tmpdirname: str) -> pd.DataFrame:
         fns = glob.glob(os.path.join(tmpdirname, "*.csv"))
         assert len(fns) == 1
         with open(fns[0], "rb") as f:
-            tsv = io.BytesIO(f.read().decode("UTF-16").encode("UTF-8"))
+            try:
+                # Downloaded data from Crosstab needs this
+                tsv = io.BytesIO(f.read().decode("UTF-16").encode("UTF-8"))
+            except Exception as e:
+                # Downloaded data from Data doesn't
```

Comment:
What's happening here in this try/except?

Is there a way to check if we need to decode/encode so we can handle the case directly?

Reply:
I was not able to find a way to do it and try and except is the pythonic way to do deal with this kind of uncertainty. If you have another suggestion, I am more than happy to change.

Reply:
That's ok for me!